A request to move @airbnb/node-memwatch to devDependencies in package.json

Any chance of `@airbnb/node-memwatch` being moved into a `devDependency` in `package.json`? As it's just increasing CI times for us - especially sad as it is a dev dependency. It also causes errors that get ignored in some environments (not a real bother or your problem).
I have resorted to using the `--no-optional` flag for `npm install` and `npm prune` and that does avoid the problem. Using that flag feels a little heavy handed for just your package though.

Although this isn't your problem, the errors I get look like:

    ..\src\memwatch.cc(19): fatal error C1083: Cannot open include file: 'sys/time.h': No such file or directory [...\node_modules\@airbnb\node-memwatch\build\memwatch.vcxproj]
    ..\src\heapdiff.cc(291): warning C4244: 'argument': conversion from 'unsigned __int64' to 'v8::SnapshotObjectId', possible loss of data [...\node_modules\@airbnb\node-memwatch\build\memwatch.vcxproj]
    ..\src\heapdiff.cc(303): warning C4244: 'argument': conversion from 'unsigned __int64' to 'v8::SnapshotObjectId', possible loss of data [...\node_modules\@airbnb\node-memwatch\build\memwatch.vcxproj]
    ..\node_modules\nan\nan_new.h(208): warning C4244: 'argument': conversion from 'unsigned __int64' to 'double', possible loss of data (compiling source file ..\src\heapdiff.cc) [...\node_modules\@airbnb\node-memwatch\build\memwatch.vcxproj]

I understand that you might not want dev package restores to fail, but increasing everyone else's bother isn't great either.

Anyway, just a humble request :)

--- a/.eslintignore
+++ b/.eslintignore
@@ -1,3 +1,3 @@
 coverage
 out
-test/plugins/versions
+node_modules

--- a/LICENSE-3rdparty.csv
+++ b/LICENSE-3rdparty.csv
@@ -16,7 +16,6 @@ require,require-in-the-middle,MIT,Copyright 2016-2018 Thomas Watson Steen
 require,safe-buffer,MIT,Copyright Feross Aboukhadijeh
 require,shimmer,BSD-2-Clause,Copyright Forrest L Norvell
 require,url-parse,MIT,Copyright 2015 Unshift.io Arnout Kazemier the Contributors
-dev,@airbnb/node-memwatch,WTFPL,Copyright Lloyd Hilaiel
 dev,amqplib,MIT,Copyright 2013-2014 Michael Bridgen
 dev,axios,MIT,Copyright 2014-present Matt Zabriskie
 dev,benchmark,MIT,Copyright 2010-2016 Mathias Bynens Robert Kieffer John-David Dalton

--- a/package.json
+++ b/package.json
@@ -10,7 +10,7 @@
     "lint": "eslint . && node scripts/check_licenses.js",
     "tdd": "node ./scripts/install_plugin_modules && mocha --watch",
     "test": "node ./scripts/install_plugin_modules && nyc --reporter text --reporter lcov mocha 'test/**/*.spec.js'",
-    "leak": "node --no-warnings ./node_modules/.bin/tape 'test/leak/**/*.js'"
+    "leak": "(cd test/leak && yarn) && NODE_PATH=./test/leak/node_modules node --no-warnings ./node_modules/.bin/tape 'test/leak/{,!(node_modules)/**/}/*.js'"
   },
   "repository": {
     "type": "git",
@@ -87,8 +87,5 @@
     "sinon": "^4.2.1",
     "sinon-chai": "^2.14.0",
     "tape": "^4.9.1"
-  },
-  "optionalDependencies": {
-    "@airbnb/node-memwatch": "^1.0.2"
   }
 }

--- a/scripts/check_licenses.js
+++ b/scripts/check_licenses.js
@@ -6,9 +6,9 @@ const readline = require('readline')
 const pkg = require(path.join(__dirname, '..', '/package.json'))
 
 const filePath = path.join(__dirname, '..', '/LICENSE-3rdparty.csv')
-const deps = Object.keys(pkg.dependencies)
-  .concat(Object.keys(pkg.devDependencies))
-  .concat(Object.keys(pkg.optionalDependencies))
+const deps = Object.keys(pkg.dependencies || {})
+  .concat(Object.keys(pkg.devDependencies || {}))
+  .concat(Object.keys(pkg.optionalDependencies || {}))
   .sort()
 
 let index = 0

--- a/test/leak/package.json
+++ b/test/leak/package.json
@@ -1,0 +1,9 @@
+{
+  "name": "leak",
+  "version": "1.0.0",
+  "license": "BSD-3-Clause",
+  "private": true,
+  "dependencies": {
+    "@airbnb/node-memwatch": "^1.0.2"
+  }
+}

